auto_submit/lib/validations/approval.dart

```diff
@@ -23,19 +23,55 @@ class Approval extends Validation {
     final String authorAssociation = pullRequest.authorAssociation!;
     final String? author = pullRequest.author!.login;
     final List<ReviewNode> reviews = pullRequest.reviews!.nodes!;
-    final bool approved = config.rollerAccounts.contains(author) ||
-        _checkApproval(
-          author,
-          authorAssociation,
-          reviews,
-        );
-
-    const String message = '- Please get at least one approved review if you are already '
-        'a member or two member reviews if you are not a member before re-applying this '
-        'label. __Reviewers__: If you left a comment approving, please use '
-        'the "approve" review action instead.';
+
+    bool approved = false;
+    String message = '';
+    if (config.rollerAccounts.contains(author)) {
+      approved = true;
+      log.info('PR approved by roller account: $author');
+      return ValidationResult(approved, Action.REMOVE_LABEL, '');
+    } else {
+      final Approver approver = Approver(author, authorAssociation, reviews);
+      approver.computeApproval();
+      approved = approver.approved;
+      log.info(
+          'PR approved $approved, approvers: ${approver.approvers}, remaining approvals: ${approver.remainingReviews}, request authors: ${approver.changeRequestAuthors}');
+      final String appvd = approved
+          ? 'This PR has met approval requirements for merging.\n'
+          : 'This PR has not met approval requirements for merging. You have project association $authorAssociation and need ${approver.remainingReviews} more review(s) in order to merge this PR.\n';
+
+      message = approved
```

Comment:
Shall we add the case with `CHANGES_REQUESTED` to the message?

Reply:
Okay I have updated the message if CHANGES_REQUESTED has been found. It can say you need 0 more reviews but will state changes have been requested and to make those in the event you get enough approvals but one person requests changes.